go-controller/pkg/util/ovs.go

```diff
@@ -202,12 +214,34 @@ func runOVNretry(cmdPath string, args ...string) (*bytes.Buffer, *bytes.Buffer,
 	}
 }
 
+// when ovn-nbctl is running in a "daemon mode", the user first starts
+// ovn-nbctl running in the background and afterward uses the daemon to execute
+// operations. The client needs to use the control socket and set the path to the
+// control socket in environment variable OVN_NB_DAEMON
+func getNbctlSockPathAsEnvVars() []string {
```

Comment:
Can we change this into something like:

`func getNbctlEnvAndArgs(scheme OvnDBScheme) ([]string, []string, error) {`

and it will pass back <cmdArgs, envVars, error>

RunOVNNbctlUnix() would pass in `OvnDBSchemeUnix` (clearly) and RunOVNNbctlWithTimeout() would pass in `config.OvnNorth.Scheme`. And then move the `if len(envVars) == 0 {` bits from RunOVNNbctlWithTimeout() into this function.

This way we can use the function in both places to deliver the right arguments and env vars and keep all the special casing logic for daemon mode in one place?

Reply:
@dcbw If the concern is that I check for `config.Default.NbctlDaemonMode` at several places, then I can remove the check in RunOVNNbctlWithTimeout(). That check is redundant and is not needed. Not sure, why I added :).

Following, in RunOVNNbctlWithTimeout() 
```
	if config.Default.NbctlDaemonMode {
		envVars = getNbctlSockPathAsEnvVars()
	}
```
can just be
```
     envVars = getNbctlSockPathAsEnvVars()
```

This will be similar to what you are asking, except that we get just the envVars from the function.

Reply:
@girishmg yeah if we can remove the check that would be nice.